### PR TITLE
Annotate Fidesops with Fidesctl

### LINKED ISSUE #1065: Annotate Fidesops with Fidesctl
### Is your feature request related to a specific problem?

Fidesops should be annotated with fidesctl

### PR DESCRIPTION
**Pull request by sanders41**
# Purpose

Annotate fidesops with fidesctl

# Changes
- Added fidesops_dataset.yaml file
- Added fidesops_system.yaml file

# Checklist
- [x] Update [`CHANGELOG.md`](https://github.com/ethyca/fidesops/blob/main/CHANGELOG.md) file
  - [x] Merge in main so the most recent `CHANGELOG.md` file is being appended to
  - [x] Add description within the `Unreleased` section in an appropriate category. Add a new category from the list at the top of the file if the needed one isn't already there.
  - [x] Add a link to this PR at the end of the description with the PR number as the text. example: [#1](https://github.com/ethyca/fidesops/pull/1)
- [ ] Applicable documentation updated (guides, quickstart, postman collections, tutorial, fidesdemo, [database diagram](https://github.com/ethyca/fidesops/blob/main/docs/fidesops/docs/development/update_erd_diagram.md).
- If docs updated (select one):
  - [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
- [ ] Good unit test/integration test coverage
- [ ] This PR contains a DB migration. If checked, the reviewer should confirm with the author that the [down_revision correctly references the previous migration](https://ethyca.github.io/fidesops/development/contributing_details/#alembic-migrations) before merging
- [ ] The `Run Unsafe PR Checks` label has been applied, and checks have passed, if this PR touches any external services

# Ticket

Fixes #1065

### CODE REVIEW

File: fides_resources/fidesops_system.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+dataset:
+- fides_key: fidesops_system
+  name: Fidesops
+  description: Privacy Request Fufillment
+  system_type: Application
+  privacy_declarations:
```
[the remark this answers is not in the document]
Comment (NevilleS) [reply fides_resources/fidesops_system.yaml:6]: Note: while I prefer using two datasets here personally, I know that `fidesctl scan dataset db` expects datasets to be grouped by schema, not subdivided into logical datasets, so we wouldn't be able to use that in CI. I think it's probably better to have the CI check for 100% coverage, than the slight improvement to the declarations... but open to what you think is the best decision.

**Review by NevilleS — CHANGES_REQUESTED** [review]
note: I (obviously) have strong opinions about privacy, so I love to leave reviews on these types of PRs!

I think this is very thorough on the DB annotations, so thanks for doing that. I have a few comments there, but they are mainly styistic.

For the overall system declarations though, I'd like to see this project declare the following resources:

1. System: Fidesops Webserver
2. System: Postgres Database Server
3. System: Redis Database Server
4. Dataset: Fidesops Application Database Dataset
5. Dataset: Fidesops Identity Cache Dataset

This would be more comprehensive and accurate from afar. While your annotations aren't wrong (they are a logical model!) I'd like to list those dependencies in the declaration so we can use this as an example for users.

**Comment by ThomasLaPiana** [comment]
@sanders41 this should also include a CI check to make sure these stay in-sync with the actual database, see fidesctl for an example

**Comment by ThomasLaPiana** [comment]
I'm also curious, did you use the auto-generate tool? If not, the scan will fail because it will look for schemas/tables with matching names. The easiest flow is definitely to auto-generate the datasets and then tweak them from there

**Comment by sanders41** [comment]
I got the db dataset and system updated. Still working on a couple of the other parts.

**Comment by ThomasLaPiana** [comment]
I moved this back into draft state while work is ongoing otherwise I'll instinctively keep reviewing it

**Pull request closed without merging**